flox/core.py

```diff
@@ -1271,18 +1224,12 @@ def dask_groupby_agg(
         token=f"{name}-chunk-{token}",
     )
 
-    if split_out > 1:
-        intermediate, group_chunks = split_blocks(
-            applied, split_out, expected_groups, split_name=f"{name}-split-{token}"
-        )
-    else:
-        intermediate = applied
-        if expected_groups is None:
-            if is_duck_dask_array(by_input):
-                expected_groups = None
-            else:
-                expected_groups = _get_expected_groups(by_input, sort=sort)
-        group_chunks = ((len(expected_groups),) if expected_groups is not None else (np.nan,),)
+    if expected_groups is None:
+        if is_duck_dask_array(by_input):
+            expected_groups = None
+        else:
+            expected_groups = _get_expected_groups(by_input, sort=sort)
+    group_chunks = ((len(expected_groups),) if expected_groups is not None else (np.nan,),)
```

Comment:
```suggestion
    group_chunks: tuple[tuple[float, ...]] = ((len(expected_groups),) if expected_groups is not None else (np.nan,),)
```

Try explicitly defining how `group_chunks` is supposed to look like. 

mypy get's the type hint from the first time a variable is defined, it's not clear from this line that it can be multiple floats like later on line 1302.

```
flox/core.py:1302: error: Incompatible types in assignment (expression has type "Tuple[Tuple[int, ...]]", variable has type "Tuple[Tuple[float]]")  [assignment]
Found 2 errors in 1 file (checked 10 source files)
flox/core.py:1322: error: Incompatible types in assignment (expression has type "Tuple[Tuple[int, ...]]", variable has type "Tuple[Tuple[float]]")  [assignment]
```

Reply:
oh right. it's only ever float in that one clase and `tuple[tuple[int]]` otherwise

Reply:
For tuples you have to add `...` to imply multiple elements and those loops in 1302 will add multiple ints. 
`np.nan` is a float so the entire thing then becomes float: `tuple[tuple[float, ...]]`